packages/iris-grid/src/PartitionSelectorSearch.tsx

```diff
@@ -275,19 +281,26 @@ class PartitionSelectorSearch<T> extends Component<
   }
 
   updateFilter(): void {
-    const { dh, initialPageSize, table } = this.props;
+    const { initialPageSize, table } = this.props;
     const { text } = this.state;
     const filterText = text.trim();
     const filters = [];
     if (filterText.length > 0) {
       const column = table.columns[0];
-      const filter = column
-        .filter()
-        .invoke(
-          'matches',
-          dh.FilterValue.ofString(`(?s)(?i).*\\Q${filterText}\\E.*`)
+      if (
+        column.type === 'java.lang.String' ||
+        column.type === 'char' ||
+        column.type === 'java.lang.Boolean' ||
+        !Number.isNaN(Number(filterText))
+      ) {
+        const filter = this.tableUtils.makeQuickFilterFromComponent(
+          column,
+          column.type === 'java.lang.String' ? `~${filterText}` : filterText
         );
-      filters.push(filter);
+        if (filter) {
```

Comment:
If `filter == null`, we should log an error at the least; I think it even makes sense to just `throw` here (since the filter should _never_ by null, otherwise we screwed something up).